modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java

```diff
@@ -2088,6 +2277,112 @@ private void _testPostObjectDefinitionWithSystemAggregationObjectField()
+			new WorkflowDefinitionLink() {
+				{
+					groupExternalReferenceCode = StringPool.BLANK;
+					workflowDefinitionName = workflowDefinition1.getName();
+				}
+			};
+
+		ObjectDefinition objectDefinition = randomObjectDefinition();
+
+		objectDefinition.setWorkflowDefinitionLinks(
+			new WorkflowDefinitionLink[] {workflowDefinitionLink1});
+
+		objectDefinition = _addObjectDefinition(objectDefinition);
+
+		com.liferay.portal.kernel.model.WorkflowDefinitionLink
+			serviceBuilderWorkflowDefinitionLink =
+				_workflowDefinitionLinkLocalService.getWorkflowDefinitionLink(
+					TestPropsValues.getCompanyId(), 0,
+					objectDefinition.getClassName(), 0, 0, false);
+
+		Assert.assertEquals(
+			workflowDefinition1.getName(),
+			serviceBuilderWorkflowDefinitionLink.getWorkflowDefinitionName());
+
+		// Site scope
+
+		Group group1 = GroupTestUtil.addGroup();
+
+		objectDefinition = randomObjectDefinition();
+
+		objectDefinition.setScope(ObjectDefinitionConstants.SCOPE_SITE);
+
+		workflowDefinitionLink1.setGroupExternalReferenceCode(
+			group1.getExternalReferenceCode());
+
+		Group group2 = GroupTestUtil.addGroup();
+
+		String content = workflowDefinition1.getContentAsXML();
+
+		WorkflowDefinition workflowDefinition2 =
+			_workflowDefinitionManager.deployWorkflowDefinition(
+				null, TestPropsValues.getCompanyId(),
+				TestPropsValues.getUserId(), RandomTestUtil.randomString(),
+				RandomTestUtil.randomString(), content.getBytes());
+
+		WorkflowDefinitionLink workflowDefinitionLink2 =
+			new WorkflowDefinitionLink() {
+				{
+					groupExternalReferenceCode =
+						group2.getExternalReferenceCode();
+					workflowDefinitionName = workflowDefinition2.getName();
+				}
+			};
+
+		objectDefinition.setWorkflowDefinitionLinks(
+			new WorkflowDefinitionLink[] {
+				workflowDefinitionLink1, workflowDefinitionLink2
+			});
+
+		objectDefinition = _addObjectDefinition(objectDefinition);
+
+		List<com.liferay.portal.kernel.model.WorkflowDefinitionLink>
+			serviceBuilderWorkflowDefinitionLinks =
+				_workflowDefinitionLinkLocalService.getWorkflowDefinitionLinks(
+					TestPropsValues.getCompanyId(),
+					objectDefinition.getClassName());
+
+		WorkflowDefinitionLink[] workflowDefinitionLinks =
+			TransformUtil.transformToArray(
+				serviceBuilderWorkflowDefinitionLinks,
+				serviceBuilderWorkflowDefinitionLinkIterator -> {
+					Group group = _groupLocalService.getGroup(
+						serviceBuilderWorkflowDefinitionLinkIterator.
+							getGroupId());
+
+					return new WorkflowDefinitionLink() {
+						{
+							groupExternalReferenceCode =
+								group.getExternalReferenceCode();
+							workflowDefinitionName =
+								serviceBuilderWorkflowDefinitionLinkIterator.
+									getWorkflowDefinitionName();
+						}
+					};
+				},
+				WorkflowDefinitionLink.class);
+
+		Assert.assertEquals(
+			new HashSet<>(
+				Arrays.asList(objectDefinition.getWorkflowDefinitionLinks())),
+			new HashSet<>(Arrays.asList(workflowDefinitionLinks)));
```

Comment:
In `Assert.assertEquals`, the first parameter should be the expected value and the second the actual. So `workflowDefinitionLinks` should go first, and `objectDefinition` second.